addons/web/static/src/core/template_inheritance.js

```diff
@@ -0,0 +1,244 @@
+    }
+}
+
+function getRoot(node) {
+    while (node.parentElement) {
+        node = node.parentElement;
+    }
+    return node;
+}
+
+const HASCLASS_REGEXP = /hasclass\(([^)]*)\)/g;
+function getXpath(node) {
+    const xpath = node.getAttribute("expr");
+    // hasclass does not exist in XPath 1.0 but is a custom function defined server side (see _hasclass) usable in lxml.
+    // Here we have to replace it by a complex condition (which is not nice).
+    return xpath.replaceAll(HASCLASS_REGEXP, (_, capturedGroup) => {
+        return capturedGroup
+            .split(",")
+            .map((c) => `contains(concat(' ', @class, ' '), ' ${c.trim().slice(1, -1)} ')`)
+            .join(" and ");
+    });
+}
+
+function getNode(source, node) {
+    const root = getRoot(source);
+    const doc = new Document();
+    doc.appendChild(root); // => root is the documentElement of its ownerDocument
+    if (node.tagName === "xpath") {
+        const xpath = getXpath(node);
+        const result = doc.evaluate(xpath, root, null, XPathResult.FIRST_ORDERED_NODE_TYPE);
+        return result.singleNodeValue;
+    }
+    for (const elem of root.querySelectorAll(node.tagName)) {
+        if (
+            [...node.attributes].every(
+                ({ name, value }) => name === "position" || elem.getAttribute(name) === value
+            )
+        ) {
+            return elem;
+        }
+    }
+    return null;
+}
+
+function getNodes(target, node) {
+    const nodes = [];
+    for (const childNode of node.childNodes) {
+        if (childNode.tagName === "xpath" && childNode.getAttribute?.("position") === "move") {
+            const node = getNode(target, childNode);
+            removeNode(node);
+            nodes.push(node);
+        } else {
+            nodes.push(childNode);
+        }
+    }
+    return nodes;
+}
+
+function mergeTextNodes(first, second, trimEnd = true) {
+    first.data = (trimEnd ? first.data.trimEnd() : first.data) + second.data;
+    second.remove();
+}
+
+function removeNode(node) {
+    const { nextSibling, previousSibling } = node;
+    node.remove();
+    if (nextSibling?.nodeType === Node.TEXT_NODE && previousSibling?.nodeType === Node.TEXT_NODE) {
+        mergeTextNodes(
+            previousSibling,
+            nextSibling,
+            previousSibling.parentElement.firstChild === previousSibling
+        );
+    }
+}
+export function applyInheritance(source, operations, url = "") {
+    for (const operation of operations.children) {
+        const target = getNode(source, operation);
+        if (!target) {
+            throw new Error(`Element '${operation.outerHTML}' cannot be located in parent view`);
+        }
+        const position = operation.getAttribute("position") || "inside";
+
+        if (odoo.debug && url) {
+            const attributes = [...operation.attributes].map(
+                ({ name, value }) =>
+                    `${name}=${JSON.stringify(name === "position" ? position : value)}`
```

Comment:
still not sure why we special-case position here.

Reply:
my mistake. The old code did ensure that position is found first even if it is not an attribute in the original template (there is a default position). With what I did, position might not be in operation.attributes and won't be found at all -> I have to restore what I did before or drop this thing as unimportant